### PR TITLE
Removed deprecated ProcessTreeKiller

### PR DESCRIPTION
**Pull request by StefanSpieker**
Removed deprecated `ProcessTreeKiller` and renamed test
According to [deprecated-and-unused](https://ci.jenkins.io/job/Infra/job/usage-in-plugins/job/master/lastSuccessfulBuild/artifact/output/deprecated-and-unused.html)
and [GitHub search on org](https://github.com/search?q=org%3Ajenkinsci+ProcessTreeKiller&type=Code)
it is unused.

### Proposed changelog entries

* Internal: Removed deprecated and unused `ProcessTreeKiller` class

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by stellargo — APPROVED** [review]

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 1]
```diff
@@ -1979,9 +1979,6 @@ Object writeReplace() {
      * This feature involves some native code, so we are allowing the user to disable this
      * in case there's a fatal problem.
      *
-     * <p>
-     * This property supports two names for a compatibility reason.
      */
-    public static boolean enabled = !SystemProperties.getBoolean(ProcessTreeKiller.class.getName()+".disable")
-                                 && !SystemProperties.getBoolean(ProcessTree.class.getName()+".disable");
+    public static boolean enabled = !SystemProperties.getBoolean(ProcessTree.class.getName()+".disable");
```
Comment (daniel-beck) [inline core/src/main/java/hudson/util/ProcessTree.java:1983]: Would revert this. There's no real reason for this change; just type out the obsolete class name.

**Review by daniel-beck — CHANGES_REQUESTED** [review]
Requesting a change for the compatibility system property, otherwise looks fine.

File: core/src/main/java/hudson/util/ProcessTree.java (hunk 1 above)
Comment (StefanSpieker) [reply core/src/main/java/hudson/util/ProcessTree.java]: Thanks for pointing that solution out. Changed it as requested.

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 2]
```diff
@@ -1980,5 +1980,6 @@ Object writeReplace() {
      * in case there's a fatal problem.
      *
      */
-    public static boolean enabled = !SystemProperties.getBoolean(ProcessTree.class.getName()+".disable");
+    public static boolean enabled = !SystemProperties.getBoolean("ProcessTreeKiller.disable")
```
Comment (timja) [inline core/src/main/java/hudson/util/ProcessTree.java:1983]: This should have the full package name https://www.jenkins.io/doc/book/managing/system-properties/#hudson-util-processtree-disable

Comment (StefanSpieker) [reply core/src/main/java/hudson/util/ProcessTree.java]: Ooops.

**Review by timja — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Pull request merged**